### PR TITLE
Add Slashings Into Blocks from Pool

### LINKED ISSUE #2766: Handle Attester and Proposer Slashings in Operations Service
Similar to attestations, and deposits, we should implement retrieval methods in the operation service for beacon block operations such as slashing and exit. These methods will be used for proposer RPC service to construct to-be-retrieved block

### PR DESCRIPTION
**Pull request by rauljordan**
Resolves #2766

---

# Description

**Write a summary of the changes you are making**

This PR finally includes all pending slashing objects from the slashings pool into beacon blocks once proposers request their data from the beacon node.

### CODE REVIEW

File: shared/testutil/block.go [hunk 1]
```diff
@@ -155,52 +155,149 @@ func GenerateFullBlock(
 	return &ethpb.SignedBeaconBlock{Block: block, Signature: signature.Marshal()}, nil
 }
 
+// GenerateProposerSlashingForValidator for a specific validator index.
+func GenerateProposerSlashingForValidator(
```
Comment (0xKiwi) [inline shared/testutil/block.go]: Do you mind adding a test for these new functions?

File: shared/testutil/block.go [hunk 2]
```diff
@@ -155,52 +155,149 @@ func GenerateFullBlock(
...
-				Slot:     bState.Slot(),
-				BodyRoot: []byte{0, 2, 0},
-			},
-		}
-		root, err = ssz.HashTreeRoot(header2.Header)
-		if err != nil {
-			return nil, err
+// GenerateAttesterSlashingForValidator for a specific validator index.
+func GenerateAttesterSlashingForValidator(
+	bState *stateTrie.BeaconState,
```
Comment (shayzluf) [inline shared/testutil/block.go]: i think attester slashing is much less restrictive.
maybe you dont need the  assignments and committees 
if we create indexed attestation instead and sign a slashable data it should be enough to be slashable
https://github.com/prysmaticlabs/prysm/blob/ecfd7bdfa118fac1224e4d9de559fd358c3b0fb6/beacon-chain/core/blocks/block_operations.go#L500
anyway your way also works

**Comment by rauljordan** [comment]
Thanks @shayzluf addressed, also @0xKiwi these are already tested because theyre called as inner functions in `generateProposerSlashings` and `generateAttesterSlashings`

File: beacon-chain/rpc/validator/proposer_test.go [hunk 3]
```diff
@@ -90,6 +93,26 @@ func TestGetBlock_OK(t *testing.T) {
...
+	proposerSlashing, err := testutil.GenerateProposerSlashingForValidator(
+		beaconState,
+		privKeys[0],
+		0, /* validator index */
+	)
+	if err := proposerServer.SlashingsPool.InsertProposerSlashing(beaconState, proposerSlashing); err != nil {
+		t.Fatal(err)
+	}
+
+	// We include max attester slashings in the pool.
```
Comment (shayzluf) [inline beacon-chain/rpc/validator/proposer_test.go]:
```suggestion
	// We include max attester slashings which is currently 1 in the pool.
```

File: beacon-chain/rpc/validator/proposer_test.go [hunk 4]
```diff
@@ -90,6 +93,26 @@ func TestGetBlock_OK(t *testing.T) {
 		Graffiti:     graffiti[:],
 	}
 
+	// We include max proposer slashings in the pool.
```
Comment (shayzluf) [inline beacon-chain/rpc/validator/proposer_test.go]:
```suggestion
	// We include max proposer slashings which is currently 1 in the pool.
```

**Comment by shayzluf** [comment]
lgtm

**Pull request merged**